src/lib.rs

```diff
@@ -172,27 +173,65 @@ mod device_create_opt {
     }
 }
 
+mod policy_grant{
+    use super::*;
+    pub fn create(cat:Option<Category>, sens:Option<Sensitivity>, sub: Option<DataSubject>, sub_id: Option<UserId>) -> PolicyGrant {
+        PolicyGrant::new(cat, sens, sub, sub_id)
+    }
+    pub fn category(p: &PolicyGrant) -> Option<Category> {p.category().cloned()}
+    pub fn sensitivity(p: &PolicyGrant) -> Option<Sensitivity> {p.sensitivity().cloned()}
+    pub fn data_subject(p: &PolicyGrant) -> Option<DataSubject> {p.data_subject().cloned()}
+    pub fn substitute_id(p: &PolicyGrant) -> Option<UserId> {p.substitute_user().cloned()}
+}
+
+mod category{
+        use super::*;
+
+    pub fn validate(s: &str) -> Result<Category, String> {Ok(s.try_into()?)}
+    pub fn value(c: &Category) -> String{c.inner().to_string()}
+}
+
+mod sensitivity{
+        use super::*;
+
+    pub fn validate(s: &str) -> Result<Sensitivity, String> {Ok(s.try_into()?)}
+    pub fn value(s: &Sensitivity) -> String{s.inner().to_string()}
+}
+
+mod data_subject{
+        use super::*;
+
+    pub fn validate(s: &str) -> Result<DataSubject, String> { Ok(s.try_into()?)}
+    pub fn value(d: &DataSubject) -> String{d.inner().to_string()}
+}
+
 mod document_create_opt {
     use super::*;
-    use ironoxide::document::DocumentEncryptOpts;
+    use itertools::EitherOrBoth;
+    use ironoxide::document::{ExplicitGrant,DocumentEncryptOpts};
     pub fn create(
         id: Option<DocumentId>,
         name: Option<DocumentName>,
         grant_to_author: bool,
         user_grants: Vec<UserId>,
         group_grants: Vec<GroupId>,
+        policy_grant: Option<PolicyGrant>
     ) -> DocumentEncryptOpts {
-        let users_and_groups = user_grants
+        let users_and_groups: Vec<UserOrGroup> = user_grants
             .into_iter()
             .map(|u| UserOrGroup::User { id: u })
             .chain(
                 group_grants
                     .into_iter()
                     .map(|g| UserOrGroup::Group { id: g }),
-            )
-            .collect();
-
-        DocumentEncryptOpts::new(id, name, grant_to_author, users_and_groups)
+            ).collect();
+
+        let explicit = ExplicitGrant::new(grant_to_author, &users_and_groups[..]);
```

Comment:
Note that this is inefficient because it copies immediately, but the struct isn't exposed and the price we're paying should be relatively small (in terms of memory).

Reply:
Maybe I'm missing the point, but can't you just pass `&users_and_groups`?

Reply:
I didn't think I could because it took a slice. Either way it's still copying back into another vec.